lib/c-api/src/deprecated/import/mod.rs

```diff
@@ -634,26 +634,27 @@ pub unsafe extern "C" fn wasmer_import_func_params_arity(
 }
 
 /// struct used to pass in context to functions (which must be back-patched)
-#[derive(Debug, Default)]
-pub(crate) struct UnsafeMutableEnv {
+
+#[derive(Debug, Default, WasmerEnv)]
+pub(crate) struct LegacyEnv {
     pub(crate) instance_ptr: Option<NonNull<CAPIInstance>>,
 }
 
-impl UnsafeMutableEnv {
+impl LegacyEnv {
     pub(crate) fn ctx_ptr(&self) -> *mut CAPIInstance {
         self.instance_ptr
             .map(|p| p.as_ptr())
             .unwrap_or(ptr::null_mut())
     }
 }
 
-/// struct used to hold on to `UnsafeMutableEnv` pointer as well as the function.
-/// we need to do this to initialize the context ptr inside of `UnsafeMutableEnv` when
+/// struct used to hold on to `LegacyEnv` pointer as well as the function.
+/// we need to do this to initialize the context ptr inside of `LegacyEnv` when
 /// instantiating the module.
 #[derive(Debug)]
 pub(crate) struct FunctionWrapper {
     pub(crate) func: NonNull<Function>,
-    pub(crate) legacy_env: NonNull<UnsafeMutableEnv>,
+    pub(crate) legacy_env: NonNull<LegacyEnv>,
```

Comment:
I actually like more `UnsafeMutableLegacyEnv`. Verbose but it conveys the meaning

Reply:
So this change is actually fixing an issue from the find and replace done on in another PR.  It should really be split out and merged to master separately, but I accidentally renamed a struct with the same name in another place by doing bulk find and replace, this just corrects that mistake